test/suites/network_ovn.sh

```diff
@@ -0,0 +1,241 @@
+  [ "$(ovn-nbctl get logical_switch_port "${router_external_switch_port_name}" options:nat-addresses)" = "router" ]
+
+  # Check internal switch port settings (router).
+  router_internal_switch_port_name="${internal_switch_name}-lsp-router"
+  [ "$(ovn-nbctl get logical_switch_port "${router_internal_switch_port_name}" type)" = "router" ]
+  [ "$(ovn-nbctl get logical_switch_port "${router_internal_switch_port_name}" options:router-port)" = "${internal_router_port_name}" ]
+  [ "$(ovn-nbctl get logical_switch_port "${router_internal_switch_port_name}" options:nat-addresses)" = "router" ]
+
+  # Check port group settings.
+  port_group_name="lxd_net${ovn_network_id}"
+  [ "$(ovn-nbctl get port_group "${port_group_name}" external_ids:lxd_project_id)" = '"1"' ]
+  [ "$(ovn-nbctl get port_group "${port_group_name}" external_ids:lxd_switch)" = "${internal_switch_name}" ]
+
+  # Check address sets.
+  address_set_ipv4_name="${port_group_name}_routes_ip4"
+  [ "$(ovn-nbctl get address_set "${address_set_ipv4_name}" addresses | jq -er '.[0]')" = "10.24.140.0/24" ]
+
+  address_set_ipv6_name="${port_group_name}_routes_ip6"
+  [ "$(ovn-nbctl get address_set "${address_set_ipv6_name}" addresses | jq -er '.[0]')" = "fd42:bd85:5f89:5293::/64" ]
+
+  # Launch an instance on the OVN network and assert configuration changes.
+  ensure_import_testimage
+  lxc launch testimage c1 --network "${ovn_network}"
+
+  # Check that this created the expected number of entries.
+  tables["DNS"]=$((tables["DNS"]+1))
+  tables["Logical_Switch_Port"]=$((tables["Logical_Switch_Port"]+1))
+  assert_row_count
+
+  c1_mac_address="$(lxc query /1.0/instances/c1 | jq -er '.config."volatile.eth0.hwaddr"')"
+  c1_uuid="$(lxc query /1.0/instances/c1 | jq -er '.config."volatile.uuid"')"
+  c1_internal_switch_port_name="${chassis_group_name}-instance-${c1_uuid}-eth0"
+
+  # Busybox test image won't bring up the interfaces by itself. Get the address and bring it up.
+  c1_ipv4_address="$(ovn-nbctl get logical_switch_port "${c1_internal_switch_port_name}" dynamic_addresses | tr -d '"' | cut -d' ' -f 2)"
+  c1_ipv6_address="$(ovn-nbctl get logical_switch_port "${c1_internal_switch_port_name}" dynamic_addresses | tr -d '"' | cut -d' ' -f 3)"
+  lxc exec c1 -- ip -4 addr add "${c1_ipv4_address}/32" dev eth0
+  lxc exec c1 -- ip -6 addr add "${c1_ipv6_address}/64" dev eth0
+  lxc exec c1 -- ip -4 route add default dev eth0
+  lxc exec c1 -- ip -6 route add default dev eth0
+
+  # Should now be able to get the same IP addresses from the instance state.
+  [ "$(lxc query /1.0/instances/c1?recursion=1 | jq -er '.state.network.eth0.addresses | .[] | select(.family == "inet").address')" = "${c1_ipv4_address}" ]
+  [ "$(lxc query /1.0/instances/c1?recursion=1 | jq -er '.state.network.eth0.addresses | .[] | select(.family == "inet6" and .scope == "global").address')" = "${c1_ipv6_address}" ]
```

Comment:
@tomponline @simondeziel I had a test failure occur in CI when running:
```
lxc exec c1 -- ip -6 addr add "${c1_ipv6_address}/64" dev eth0
```
which returned `RTNETLINK answers: File exists`.

After testing locally, I've noticed that the IPv6 address does appear by itself after waiting a bit. Can you tell me why this happens?

I've updated the test to wait until the IPv6 address is reported by LXD, then validate that it is the same as the expected value (from dynamic addresses of the internal switch port). I'll add the explanation to a comment above the while loop.

Reply:
IPv6 addresses can take a bit of time to appear due to DAD (Duplicate Address Detection) happening.

Reply:
But why does the IPv6 address come up and the IPv4 address does not? E.g. why do I need `ip -4 addr/route` add but not `ip 6 addr/route add`?

Reply:
Could it be SLAAC assigned? I don't remember from reading your doc if OVN implements SLAAC :)

Reply:
So with IPv6, you have SLAAC which is a mechanism for automatic IP allocation/assignment coupled with DAD (introducing some delay). The "client-side" of this is is enabled by default and done by the kernel on new ifaces. There is no such thing with IPv4 so maybe that's why?

Reply:
Ok great that makes sense (after googling SLAAC and DAD :laughing:). I'll add some comments. I also recall seeing a `wait_for_dad` util somewhere (maybe in LXD CI?) so I can use that rather than checking for an IP and sleeping.